### ISSUE TITLE
Alert mode

### ISSUE THREAD

**Issue by razor20171** [FreshRSS/FreshRSS#2051]
Hi, is it possible to activate an alert with a keyword during a global refresh?
ex: mail, notification, ...

**Comment by Alkarex** [comment]
Hello,
Currently, what can be done is the following:

1. Make one or several searches in FreshRSS
2. The result of each of these searches can be accessed by RSS (set a token in the FreshRSS settings to make the RSS feed visible from outside without login)
![image](https://user-images.githubusercontent.com/1008324/47036700-4e534f80-d17d-11e8-9217-38552b107b9e.png)
    * See additional RSS feed parameters https://github.com/FreshRSS/FreshRSS/pull/1251
3. Use whatever RSS-to-notification service, such as [IFTTT](https://ifttt.com/feed) or [alternatives](https://alternativeto.net/software/ifttt/?license=opensource&platform=self-hosted), giving your FreshRSS search result RSS feed as input.

In the future, a subset of those notification features could potentially be integrated in FreshRSS itself. (Probably only simple e-mails, as it would be out of scope to duplicate a full event-processing / notification system)

Related ticket: https://github.com/FreshRSS/FreshRSS/issues/1718

**Comment by Legalien** [comment]
Let me just say. This would be a nice addition to FreshRSS.

IFTTT can be used but you can't control how often IFTTT checks for updates, so you really don't have full control over the situation.

I'd code this myself, but I don't have the necessary skills to do that, not even close.

Thanks for the alternatives link Alkarex, I'll check it out 😃

**Comment by sinelegibus** [comment]
I use huginn (https://github.com/huginn/huginn) to inform me about new RSS entries.
First you can create an agent which fetches new rss entries. You can define how often the agent should check.
Then you can define how you will be informed when a new entry is found. I set it up that I receive an XMPP message to my phone. I defined the message to include title, url and a link to my freshrss instance.

**Comment by Alkarex** [comment]
Good idea @sinelegibus 👍 Would you mind sharing some of your code for that?

**Comment by sinelegibus** [comment]
Well, it's pretty straightforward as everything is already there in huginn:

1. Create an agent with 'Type: Rss Agent'.
In the options add your feed link:
`{
  "expected_update_period_in_days": "10",
  "clean": "false",
  "url": "https://myfeed.com/feed",
  "include_feed_info": "true"
}`

2. Create an agent with 'Type: Jabber Agent'.
`{
  "jabber_server": "conversations.im",
  "jabber_port": "5222",
  "jabber_sender": "XXX@conversations.im",
  "jabber_receiver": "ZZZ@conversations.im",
  "jabber_password": "YYY",
  "message": "New RSS entry: my.freshrss.com\r\n{{title}}\r\n{{url}}",
  "connect_to_receiver_radio": "true",
  "connect_to_receiver": "true",
  "expected_receive_period_in_days": "10"
}`

**Comment by Alkarex** [comment]
Thanks for the feedback @sinelegibus 👍

**Comment by Legalien** [comment]
I looked around and found something to use. This is what I'm using - [https://github.com/e-dschungel/rssgoemail](https://github.com/e-dschungel/rssgoemail)

I did see Huginn while searching for a solution, and I passed on it. I mean, look at the Huginn installation guide:

![22222222222222](https://user-images.githubusercontent.com/44560211/48070561-6c3e2e00-e1d8-11e8-9374-7d67f2b080c8.PNG)

And look at the Rssgoemail installation guide:

![1111111](https://user-images.githubusercontent.com/44560211/48070594-7eb86780-e1d8-11e8-8ec7-dbd32b499a35.PNG)

I mean, I'm not a rocket scientist man. I went with Rssgoemail, it's so much easier. I can actually see myself installing it. Huggins, where do I even begin? :)

That said... I installed Rssgoemail and I've been using it and it works great. It emails me when any of my feeds has new content on it. However, if my feeds have Facebook Posts on them then for some reason it emails me every hour. I'm trying to figure it out - [https://github.com/e-dschungel/rssgoemail/issues/3](https://github.com/e-dschungel/rssgoemail/issues/3)

Maybe it has to do with Facebook Post URLs and how each URL has that long string of characters in it, and maybe that long string of characters is dynamic and changes every half an hour or every hour, I don't know. I'm trying to figure it out. I edited the Rssgoemail code to stop using the permalink to detect new entries, and to only use the Title and Content. So I'll see if I still keep getting updates every hour because of the Facebook posts now that the Permalink isn't checked anymore. I'm testing it out now.

**Comment by sinelegibus** [comment]
Well, you missed/overlooked the easy way:

### *Docker*

*The quickest and easiest way to check out Huginn is to use the official Docker image. Have a look at the [documentation](https://github.com/huginn/huginn/blob/master/doc/docker/install.md).*

Anyway, it's always good to have more than one possibility.

**Comment by Frenzie** [comment]
I use Docker a fair bit, but calling it "easy" is a bit far-fetched even in the best case scenario (such as apt or zypper) imo. :-p

**Comment by Frenzie** [comment]
Congrats! Make it a good one. ;-)

**Comment by squromiv** [comment]
What is about [Apprise](https://github.com/caronc/apprise) or [Apprise-api](https://github.com/caronc/apprise-api) tool? It may be fine to push an alert to Apprise as soon as some "hot" feed will have new items. Services like Huginn, n8n, Node-RED etc are not so convenient for such a purpose.
